src/solvers/iterators.jl

```diff
@@ -0,0 +1,104 @@
+#
+# SweepIterator
+#
+
+mutable struct SweepIterator
+  sweep_kws
+  region_iter
```

Comment:
Maybe this should be called `current_region_iter` or something like that to indicate it is the latest region iterator.

I have to say this part confused me a bit. I see the region iterator is recreated from scratch from the `sweep_kws` at each sweep, is there a reason to store it? It seems like it could just be made and used "on the fly" at each sweep anyway.

Reply:
Yes, keeping the problem in the sweep iterator could simplify the mechanics quite a lot. I was thinking originally like "better to have fewer fields in SweepIterator, especially since the region iterator will have a copy of the problem" but I think on balance it's best not to have complicated code handling creating each region iterator.